staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/policy_decision.go

```diff
@@ -34,6 +35,16 @@ type policyDecision struct {
 	kind    policyDecisionKind
 	message string
 	reason  metav1.StatusReason
+	elapsed time.Duration
+	// denyIgnored is a flag indicating that policyDecision.kind being set to
+	// "admit" was done due to failure policy - so either there was an error
+	// while evaluating an expression, or an expression returned false,
+	// but failurePolicy was set to "ignore"
+	// this is important for metrics, since we want to differentiate between
+	// a regular "admit" (no evaluation errors/expressions failing to validate)
+	// versus an "admit" due to failure policy (some error/failure occurred, but
+	// was ignored)
```

Comment:
This is extremely confusing.
Consider having two axes:
* evaluation: error, admit, block
* action: admit, block

Reply:
Makes sense - I'll rework `policyDecisionKind` into `policyDecisionAction`  and then add a `policyDecisionEvaluation` type and add that into `policyDecision` and the other appropriate places.

Reply:
Much clearer, thanks. Can you squash?